modules/core/src/smithy4s/schema/Field.scala

```diff
@@ -65,10 +65,8 @@ final case class Field[S, A](
    * Applies a side-effecting lambda if the field is marked as required (using `smithy.api#required`),
    * OR if the field value is different from its default. Otherwise nothing happens.
    */
-  def foreachUnlessDefault(s: S)(f: A => Unit): Unit = {
-    val a = get(s)
-    if (isDefaultValue(a) && !isRequired) () else f(a)
```

Comment:
actually the reason we didn't factor it was to avoid an un-necessary allocation to `Option`, so I'm gonna ask you to revert (feel free to add a comment to point that out)

Reply:
Done, thanks for letting me know!